bundle/edu.gemini.oodb.too.url/src/main/java/edu/gemini/spdb/rapidtoo/impl/TooUpdateServiceImpl.java

```diff
@@ -237,10 +237,9 @@ private void _setTarget(ISPObservation obs, TooUpdate update)  {
         TargetEnvironment targetEnv = targetObsComp.getTargetEnvironment();
         SPTarget base = targetEnv.getBase();
 
-        base.getTarget().setName(tooTarget.getName());
-        base.getTarget().getRa().setAs(tooTarget.getRa(), CoordinateParam.Units.DEGREES);
-        base.getTarget().getDec().setAs(tooTarget.getDec(), CoordinateParam.Units.DEGREES);
-        base.getTarget().setMagnitudes(tooTarget.getMagnitudes());
+        base.setName(tooTarget.getName());
+        base.setRaDecDegrees(tooTarget.getRa(), tooTarget.getDec());
+        base.setMagnitudes(tooTarget.getMagnitudes());
         base.notifyOfGenericUpdate();
```

Comment:
Correct, we don't need this. I have been deleting them but I missed this one. I'll do another sweep.

You're also correct that there will be more events now than before. `SPTarget` may end up with a  `.modify(f: Target => Target)` or something so we can do several updates atomically. In any case I'll try out TPE dragging and a few other things and see if the extra events are gumming things up.